src/Storages/MergeTree/MergedBlockOutputStream.cpp

```diff
@@ -207,6 +224,8 @@ void MergedBlockOutputStream::writeImpl(const Block & block, const IColumn::Perm
         return;
 
     writer->write(block, permutation);
+    new_serialization_info.add(block);
```

Comment:
Why do we build this serialization separately after write?
Why it is not the same as that used fro write ...
(As I understand, we will store it in part meta)

Reply:
Because during merge some values may be deleted by TTL or collapsing algorithm, e.g. and we need to recalculate statistics about content of columns.

Reply:
I've added a flag to avoid unnecessary recalculations of `SerializationInfo`, e.g. on inserts.